### PR TITLE
Foundation laid for global Cripts

### PR DESCRIPTION
**Pull request by zwoop**
This is a beginning, there's more work to do here, looking for contributions:

1. The Bundles callbacks is untested and may need work.
2. We should add other non-HTTP txn hooks, such as the SNI hooks
3. Possibly, we should add the other HTTP-txn hooks as well (such as DNS etc.)
4. More testing / real code and uses cases.

### CODE REVIEW

**Comment by zwoop** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
[approve ci autest]

File: doc/developer-guide/cripts/cripts-global.en.rst [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+Lets look at the normal hooks that are available in Cripts. Note here that the name
+of the function dictates the underlying ATS hook.
+
+.. _cripts-global-hooks-txn-start:
+
+glb_txn_start()
+^^^^^^^^^^^^^^^
+
+The ``glb_txn_start()`` hook is called at the beginning of a transactio. This is also
```
Comment (bryancall) [inline doc/developer-guide/cripts/cripts-global.en.rst:83]: transactio -> transaction

File: doc/developer-guide/cripts/cripts-global.en.rst [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+The ``glb_read_request()`` hook is called after the client request has been read. This
+means that you can access the request headers, and the request URL. However, remap rules
+has not yet been applied, so the URL may not be the final URL, or even complete.
+
+.. _cripts-global-hooks-pre-remap:
+
+glb_pre_remap()
+^^^^^^^^^^^^^^^
+
+This hook may not be particuarly useful in Cripts, as remap rules are not applied her yet
```
Comment (bryancall) [inline doc/developer-guide/cripts/cripts-global.en.rst:101]: particuarly -> particularly

**Review by bryancall — CHANGES_REQUESTED** [review]

**Review by bryancall — APPROVED** [review]

**Pull request merged**